src/pause/138CC0.c

```diff
@@ -103,19 +103,386 @@ s32 pause_badges_scroll_offset_x(s32 x) {
+    }
+}
 
 INCLUDE_ASM(s32, "pause/138CC0", pause_badges_draw_contents);
 
-INCLUDE_ASM(s32, "pause/138CC0", pause_badges_load_badges);
+void pause_badges_load_badges(s32 onlyEquipped) {
+    PlayerData* playerData = &gPlayerData;
+    s32 numItems = 0;
+    PauseItemPage* page;
+    s32 i;
 
-INCLUDE_ASM(s32, "pause/138CC0", pause_badges_init);
+    D_80270388 = 0;
+    if (!onlyEquipped) {
+        for (i = 0; i < ARRAY_COUNT(playerData->badges); ++i) {
+            s16 badgeItemID = playerData->badges[i];
+            if (badgeItemID == 0) {
+                continue;
+            } else if (badgeItemID >= ITEM_PARTNER_ATTACK) {
+                break;
+            } else {
+                gBadgeMenuItemIDs[numItems] = badgeItemID;
+                numItems += 1;
+            }
+        }
+    } else {
+        s16* equippedBadges;
+        pause_badges_count_equipped();
+        equippedBadges = playerData->equippedBadges;
+        for (i = 0; i < ARRAY_COUNT(playerData->equippedBadges); ++i) {
+            // This can be written more clearly as equippedBadges[i],
+            // but that causes some instruction reordering
+            s16 badgeItemID = *equippedBadges;
+            if (badgeItemID != 0) {
+                gBadgeMenuItemIDs[numItems] = badgeItemID;
+                numItems += 1;
+            }
+            ++equippedBadges;
+        }
+    }
+    if (numItems == 0) {
+        gBadgeMenuItemIDs[0] = 0x7FFE;
+        numItems = 1;
+    }
+    gBadgeMenuNumItems = numItems;
+    for (i = numItems; i < ARRAY_COUNT(gBadgeMenuItemIDs); ++i) {
+        gBadgeMenuItemIDs[i] = 0x7FFF;
+    }
 
-INCLUDE_ASM(s32, "pause/138CC0", pause_badges_handle_input);
+    gBadgeMenuSelectedIndex = 0;
+    gBadgeMenuSelectedItemID = 0;
+    D_8027037C = 0;
+    D_80270394 = 0;
+    gBadgeMenuCurrentPage = 0;
+
+    page = &gBadgeMenuPages[0];
+    i = 0;
+   
+    while (1) {
+        s32 totalItems = gBadgeMenuNumItems;
+        if (i >= totalItems / 8) {
+            break;
+        }
+
+        page->listStart = i*8;
+        page->numCols = 1;
+        page->numRows = 8;
+        page->enabled = 1;
+        page->startIndex = i*8;
+        page->count = 8;
+
+        i += 1;
+        page += 1;
+    }
+
+    if (((u16) gBadgeMenuNumItems & 7) != 0) {
+        s16 count;
+        s32 menuNumItems;
+        page->listStart = i * 8;
+        page->numCols = 1;
+        page->enabled = 1;
+        page->startIndex = i * 8;
+
+        menuNumItems = gBadgeMenuNumItems;
+        count = menuNumItems - (menuNumItems / 8 * 8);
+        page->count = count;
+        page->numRows = page->count;
+        i += 1;
+        page += 1;
+    }
+
+    while (i < ARRAY_COUNT(gBadgeMenuPages)) {
+        page->enabled = 0;
+
+        ++i;
+        ++page;
+    }
+
+    gBadgeMenuTargetScrollPos = gBadgeMenuCurrentScrollPos = pause_badges_get_pos_y(0, 0);
+}
+
+void pause_badges_init(MenuPanel *panel) {
+    s32 i;
+
+    (void)pause_badges_count_all();
```

Comment:
Is there a reason the void cast is here? I think it can be removed

Reply:
Force of habit, I usually include it when a function's return value is being discarded. In this case it looks like they call count_all for its side effect of sorting the badge list, rather than to actually get the badge count